domain/user/state/shared.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"time"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/permission"
+	"github.com/juju/juju/core/user"
+	permissionstate "github.com/juju/juju/domain/permission/state"
+	usererrors "github.com/juju/juju/domain/user/errors"
+	internaldatabase "github.com/juju/juju/internal/database"
+	internaluuid "github.com/juju/juju/internal/uuid"
+)
+
+// SharedState is the shared state for the user domain. Methods which are shared
+// between multiple domains.
+// This composes the permission state as it is shared between user and
+// permission domains.
+type SharedState struct {
```

Comment:
Permission shares `AddUserPermission`, who is `AddUser`/`AddUserWithPassword` shared with?